app/services/stats/management_information/daily_report_count_generator.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+module Stats
+  module ManagementInformation
+    class DailyReportCountGenerator
```

Comment:
will have a quick look at but with only two things sharing the logic it may not be worth it (rule of three+), plus may be changing the count generator further.